lib/dialects/mysql/schema/mysql-tablecompiler.js

```diff
@@ -92,6 +92,10 @@ class TableCompiler_MySQL extends TableCompiler {
               if (column.Default !== void 0 && column.Default !== null) {
                 sql += ` DEFAULT '${column.Default}'`;
               }
+              // Add back the auto increment if the column had it, fix issue #2767
+              if (column.Extra == 'auto_increment') {
```

Comment:
I think this part is not currently covered by tests, could you add one?

Also shouldn't similar changes be done for other dialects as well?

Reply:
It is a mySQL only issue. Will add a tests for it

Reply:
@gpetrov It would be best to add a proper integration test (see test/integration2) that would try inserting something after manipulating schema and checking that autoincrement works as expected. That would cover all the DBs automatically.

Reply:
well I just extended the already available renameColumn tests - there were just fine to check for the the auto incremental afterwards.